src/py21cmfast/src/HaloBox.c

```diff
@@ -830,20 +818,20 @@ int ComputeHaloBox(double redshift, struct UserParams *user_params, struct Cosmo
             set_fixed_grids(M_min, M_cell, ini_boxes, perturbed_field, previous_spin_temp, previous_ionize_box, &hbox_consts, grids, &averages_box);
             M_turn_a_global = averages_box.m_turn_acg;
             M_turn_m_global = averages_box.m_turn_mcg;
-            get_box_averages(M_min, M_cell, M_turn_a_global, M_turn_m_global, &hbox_consts, &averages_global);
-            //This is the mean adjustment that happens in the rest of the code
-            //NOTE: in the default mode, global averages are fixed separately for minihalo and regular parts
-            #pragma omp parallel for num_threads(user_params->N_THREADS)
-            for(idx=0;idx<HII_TOT_NUM_PIXELS;idx++){
-                grids->halo_mass[idx] *= averages_global.halo_mass/averages_box.halo_mass;
-                grids->halo_stars[idx] *= averages_global.stellar_mass/averages_box.stellar_mass;
-                grids->halo_stars_mini[idx] *= averages_global.stellar_mass_mini/averages_box.stellar_mass_mini;
-                grids->halo_sfr[idx] *= averages_global.halo_sfr/averages_box.halo_sfr;
-                grids->halo_sfr_mini[idx] *= averages_global.sfr_mini/averages_box.sfr_mini;
-                grids->halo_xray[idx] *= averages_global.halo_xray/averages_box.halo_xray;
-                grids->n_ion[idx] *= averages_global.n_ion/averages_box.n_ion;
-                grids->whalo_sfr[idx] *= averages_global.fescweighted_sfr/averages_box.fescweighted_sfr;
-            }
+            // get_box_averages(M_min, M_cell, M_turn_a_global, M_turn_m_global, &hbox_consts, &averages_global);
+            // //This is the mean adjustment that happens in the rest of the code
+            // //NOTE: in the default mode, global averages are fixed separately for minihalo and regular parts
+            // #pragma omp parallel for num_threads(user_params->N_THREADS)
+            // for(idx=0;idx<HII_TOT_NUM_PIXELS;idx++){
+            //     grids->halo_mass[idx] *= averages_global.halo_mass/averages_box.halo_mass;
+            //     grids->halo_stars[idx] *= averages_global.stellar_mass/averages_box.stellar_mass;
+            //     grids->halo_stars_mini[idx] *= averages_global.stellar_mass_mini/averages_box.stellar_mass_mini;
+            //     grids->halo_sfr[idx] *= averages_global.halo_sfr/averages_box.halo_sfr;
+            //     grids->halo_sfr_mini[idx] *= averages_global.sfr_mini/averages_box.sfr_mini;
+            //     grids->halo_xray[idx] *= averages_global.halo_xray/averages_box.halo_xray;
+            //     grids->n_ion[idx] *= averages_global.n_ion/averages_box.n_ion;
```

Comment:
So, we're not doing the mean adjustment any more? If so we should just delete these lines.

Reply:
I really don't like the mean adjustments, and I've removed *some* need for it by including the Sheth-Tormen CMF a while back. This made the testing a lot easier but I'm doing some comparison runs now with/without